File: Comet/Respring/Respring.m
//
//  Respring.m
//  Comet
//
//  Created by Noah Little on 26/3/2023.
//

#import <Foundation/Foundation.h>
#import <objc/runtime.h>
#import "Respring.h"
#import "Respring-Private.h"

@implementation Respring

+ (void)execute {
    SBSRelaunchAction *action = [
        objc_getClass("SBSRelaunchAction")
        actionWithReason:@"RestartRenderServer"
        options:SBSRelaunchActionOptionsFadeToBlackTransition
        targetURL:nil
    ];
    
    FBSSystemService *service = [objc_getClass("FBSSystemService") sharedService];
    [service sendActions:[NSSet setWithObject:action] withResult:nil];
}

@end

File: Comet/Respring/Respring.swift
//
//  Respring.swift
//  Comet
//
//  Created by Noah Little on 23/4/2023.
//

import Foundation

public struct Respring {
    public static func execute() {
        let action = _SBSRelaunchAction(
            reason: "RestartRenderServer",
            options: .fadeToBlackTransition,
            targetURL: nil
        )
        
        _FBSSystemService.shared().send([action], withResult: nil)
    }
}
